### DISCUSSION TITLE
Prisma middleware doesn't work for me

### DISCUSSION THREAD

**Discussion by chonselgama** [prisma/prisma#11337] [Q&A]
Hello here, who has already used middlewares option in prismaServiceOptions. It doesn’t work for me.
I’m wondering if i should add prisma.$use at the end of middleware file before exporting.

IN MIDDLEWARE FILE :

```
import { Prisma } from "@prisma/client";
import { EntityStatusHistoryService } from "src/modules/task/services/entity-status-history.service";
import { EntityStatusService } from "src/modules/task/services/entity-status.service";
import { SaeiStatusEnum } from "src/utils/enums/saei-status.enum";
import { PrismaService } from "src/utils/prisma/prisma.service";

export function adjournApplicationStatusHistoryMiddleware(): Prisma.Middleware {
    return async function (
        params: Prisma.MiddlewareParams,
        next: (params: Prisma.MiddlewareParams) => Promise<any>,
      ): Promise<Prisma.Middleware> {
        if (params.model == 'Entity' && params.action == 'update') {
          console.log("IN FIRST CONDITION");
          const statusId = (
            await new EntityStatusService(new PrismaService()).getEntityStatus(SaeiStatusEnum.AJOURNEE)
          ).id;
          const lastEntityStatusHistory =
            await new EntityStatusHistoryService(new PrismaService()).getLastEntityStatusHistory(
              params.args.where.id,
            );
          if (
            params.args.data.entityStatusId === statusId &&
            lastEntityStatusHistory.statusId !== statusId
          ) {
            console.log("IN SECOND CONDITION");
            await new EntityStatusHistoryService(new PrismaService()).createEntityStatusHistory(statusId,params.args.where.id);
          }
        }
        return next(params);
      }
} ;
```

IN APP MODULE

```
import { HttpModule } from '@nestjs/axios';
import { Module } from '@nestjs/common';
import { ScheduleModule } from '@nestjs/schedule';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { TaskHandlerModule } from './modules/task/task.module';
import { PrismaModule } from "nestjs-prisma";
import { adjournApplicationStatusHistoryMiddleware } from './middleware/adjourn-application-status-history.middleware';

@Module({
  imports: [
    ScheduleModule.forRoot(),
    TaskHandlerModule,
    HttpModule,
    PrismaModule.forRoot({
      isGlobal: true,
      prismaServiceOptions: {
        middlewares: [adjournApplicationStatusHistoryMiddleware()],
        prismaOptions: { log: ['info'] }
      },
    }),
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
```

**Comment by janpio** [comment]
What is this `prismaServiceOptions` you are talking about? I don't think plain Prisma provides this. Is this maybe part of `nestjs-prisma` you seems to be using? Then this question is probably better asked there.

Generally you need to `prisma.$use` any middleware once to activate it.

**Reply by chonselgama** [reply]
prismaServiceOptions is part of nestjs-prisma.
At fisrt time, i used prisma.$use but after i have installed nestjs-prisma.

**Reply by pantharshit00** [reply]
Hello @chonselgama, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

Can you please share a minimal reproduction inside of a small git repository so that we can take a look here? Right now current information is not enough for us to get a reproduction.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
